src/main/resources/assets/computercraft/lua/rom/apis/window

```diff
@@ -276,6 +286,33 @@ function create( parent, nX, nY, nWidth, nHeight, bStartVisible )
         setTextColor( color )
     end
 
+    function window.setColour( colour, r, g, b )
+        if type(colour) == "table" then
+            for k,v in pairs(colour) do
+                tPalette[k] = v
+            end
+        elseif type(colour) == "number" and type(r) == "number" and g == nil and b == nil then
+            tPalette[ colour ] = { colours.rgb8( r ) }
+        elseif type(colour) == "number" and type(r) == "number" and type(g) == "number" and type(b) == "number" then
+            local tCol = tPalette[ colour ]
+            tCol[1] = r
+            tCol[2] = g
+            tCol[3] = b
+        end
+
+        if bVisible then
+            return updatePalette()
```

Comment:
Updating the whole palette is unnecessary here, just update the one colour that changed